### DISCUSSION TITLE
Problem with render material marker after upgrading app to iTwin 3.x

### DISCUSSION THREAD

**Discussion by ovaisa** [iTwin/itwinjs-core#5268] [General] [answered]
I am having a problem with RenderMaterial code after I upgraded to v3.  This code was working properly in 2.x but after I upgraded to v3 (I believe it it was 3.1x at that time - currently using 3.5.1), it started exhibiting this problem.  Any guidance on how to fix this problem would be appreciated.

The issue is that when I turn on the markers, they are drawn correctly (top image).  As soon as I move the cursor (screen is refreshed), the markers change their image as shown as in bottom image.

![pre](https://user-images.githubusercontent.com/83007711/227143060-ae993a12-9072-415b-adc4-42be73628ce1.jpg)

![post](https://user-images.githubusercontent.com/83007711/227143169-fa159289-9f60-4eb2-b65e-a4a724f7d25f.jpg)

I implemented these as markers, using code from WorldDecorationMarkers sandbox (https://www.itwinjs.org/sandboxes/ThomasDicarlo/World%20Decoration%20Markers), my requirement is identical to the example. 

Code snippet is below - addMarker() is called by iTwin when marker is to be drawn. 

```
public async addMarker(context: DecorateContext) {
   super.addMarker(context);

   const builder = context.createGraphicBuilder(GraphicType.WorldDecoration);

   const angle = context.viewport.rotation.transpose();
   angle.multiplyMatrixMatrix(Matrix3d.create90DegreeRotationAroundAxis(0));

   const point = this.worldLocation.clone();
   let decorationEllipse = Arc3d.createScaledXYColumns(
     point,
     angle.multiplyMatrixMatrix(Matrix3d.create90DegreeRotationAroundAxis(2)),
     0.8, 0.8);

   const graphicParams = new GraphicParams();
   graphicParams.material = this.material;   // created earlier by calling createMaterial() below
   builder.activateGraphicParams(graphicParams);
   builder.addArc(decorationEllipse, true, true);

   context.addDecoration(GraphicType.WorldDecoration, builder.finish());
}

public static createMaterial = async (name: string, imageURL: string): Promise<RenderMaterial> => {
  let renderMaterial: RenderMaterial;

  const image = await imageElementFromUrl(imageURL);
  if (image) {
    const textureImage: TextureImage = { source: image };
    const textureArgs: CreateTextureArgs = { image: textureImage };

    const texture = IModelApp.renderSystem.createTexture(textureArgs)!;

    const props: MaterialTextureMappingProps = {
      texture: texture,
    }
    const args: CreateRenderMaterialArgs = {
      textureMapping: props,
    }
      
    renderMaterial = IModelApp.renderSystem.createRenderMaterial(args)!;
  }

  return renderMaterial;
}
```

**Comment by pmconne** [comment, marked as answer]
You need to specify the [ownership](https://www.itwinjs.org/reference/core-frontend/rendering/createtextureargs/ownership/) of your texture.

**Reply by ovaisa** [reply]
Thank you Paul, defining ownership as "external" seems to have solved the problem, it is now working as expected.  

Appreciate the quick response!

Regards,

Ovais
